lib/message_bus/backends/active_record.rb

```diff
@@ -0,0 +1,205 @@
+
+        # @param max_backlog_age [Integer] age in seconds
+        # @return [void]
+        def expire(max_backlog_age)
+          Rails::Message.where('added_at < ?', Time.current - max_backlog_age.seconds).delete_all
+        end
+
+        # @param channel [String]
+        # @param backlog_id [Integer]
+        # @return [Array<Array<Integer, Hash>>]
+        def backlog(channel, backlog_id)
+          Rails::Message.where(channel: channel).where('id > ?', backlog_id).order(id: :asc).pluck(:id, :value)
+        end
+
+        # @param backlog_id [Integer]
+        # @return [Array<Array<Integer, String, Hash>>]
+        def global_backlog(backlog_id)
+          Rails::Message.where('id > ?', backlog_id).order(id: :asc).pluck(:id, :channel, :value)
+        end
+
+        # @param channel [String]
+        # @param id [Integer]
+        # @return [Hash, nil]
+        def get_value(channel, id)
+          Rails::Message.find_by(channel: channel, id: id)&.value
+        end
+
+        # @return [void]
+        def after_fork
+          sync do
+            @listening_on.clear
+          end
+        end
+
+        # Dangerous, truncates the message_bus table containing the backlog if it exists.
+        # @return [void]
+        def reset!
+          Rails::Message.connection.execute("TRUNCATE TABLE #{Rails::Message.table_name} RESTART IDENTITY")
+        end
+
+        # It is supposed to close all connections from connection pool. We don't need this - ActiveRecord manages
+        # connection pool
+        def destroy
+        end
+
+        # Use with extreme care, will nuke all of the data
+        # @return [void]
+        def expire_all_backlogs!
+          reset!
+        end
+
+        # @param channel [String, nil]
+        # @return [Integer]
+        def max_id(channel = nil)
+          rel = Rails::Message.all
+          rel = rel.where(channel: channel) if channel
+          rel.maximum(:id).to_i
+        end
+
+        # @param channels [String]
+        # @return [Array<Integer>]
+        def max_ids(*channels)
+          channels_info = Rails::Message.where(channel: channels).group(:channel).maximum(:id)
+          channels_info.each_with_object(Array.new(channels.size, 0)) do |(channel, max_id), result|
+            result[channels.index(channel)] = max_id
+          end
+        end
+
+        # @param channel [String]
+        # @param data [String]
+        # @return [void]
+        def publish(channel, data)
+          pub_sub_instance.publish(channel, data)
+        end
+
+        # Subscribed to the given channel. This operation blocks current thread
+        # @param channel [String]
+        # @return [void]
+        def subscribe(channel)
+          obj = Object.new
+          sync { @listening_on[channel] = obj }
+          listener = Listener.new
+          yield listener
+
+          listener.do_sub.call
+          while listening_on?(channel, obj)
+            pub_sub_instance.subscribe(channel) do |_, payload|
```

Comment:
|_, payload| -> |_channel, payload|

Reply:
There is already a local variable named `channel`. I don't want to confuse a developer who would be reading that code.